### ISSUE TITLE
error when trying to use a AWS S3 Bucket that contains a dot . in the Bucket name

### ISSUE THREAD

**Issue by maxmarines** [n8n-io/n8n#1767]
**Describe the bug**
There is error when trying to use a  AWS S3 Bucket that contains a dot . in the name

**To Reproduce**
1. Go to AWS S3 Bucket Node
2. Create Bucket with dot in name (eg mysite.com)
3. See error
"Problem executing workflow
There was a problem executing the workflow:
"Converting circular structure to JSON --> starting at object with constructor 'Object' --- property 'issuerCertificate' closes the circle""
![n8n error screen](https://user-images.githubusercontent.com/62845038/117718215-7c3d8b00-b1e4-11eb-9b4a-891b5cc6d98b.png)

**Expected behavior**
Creating the Bucket with dot in name (eg mysite.com)

**Environment (please complete the following information):**
 - OS: Ubuntu 20 LTS
 - n8n Version 0.118.1 (docker hub image)
 - Node.js Version (docker hub image)

**Comment by ivov** [comment]
@maxmarines Per the AWS sources above, path-style requests are being deprecated very soon. Since buckets with dots in their names require the user to handle their own certificate validation, it would be advisable to consider renaming the bucket.

**Comment by maxmarines** [comment]
@ivov renaming is not possible, because to use the S3 Bucket with your own domain, you must create a Bucket with dots in the name https://docs.aws.amazon.com/AmazonS3/latest/userguide/website-hosting-custom-domain-walkthrough.html

For me personally, The problem with the certificate is not important for these purposes I use by Clouflare CDN + Certificate.

**Comment by ivov** [comment]
Thank you for the clarification. Will try to take a closer look soon.

**Comment by janober** [comment]
The fix for the issue that no error got displayed got released with n8n@0.124.0

**Comment by krynble** [comment]
Hi everyone.

I was checking this issue again and also the updates from AWS. They created a post about this here: https://aws.amazon.com/pt/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/

TL;DR: the only way that currently works to handle bucket names with periods is the one being deprecated eventually.

With this in mind, we decided not to change our implementation as it will break eventually. AWS wanted to stop supporting this method on October 2020 but they are temporarily supporting this but no permanent solution was created.

Unfortunately we cannot offer a good solution as it's a problem with AWS implementation and hacking something to work now will cause problems in the future.

I will be closing this issue as no action can be taken right now.

**Comment by rootindex** [comment]
The above mentioned blog post was in [2019](https://github.com/n8n-io/n8n/issues/1767#issuecomment-896054346), as of today in 2024 the bug can still be replicated.

If you create a bucket with a dot in S3 (images.example.com), there is no way to interact with that bucket currently when trying to interact in the AWS S3 Node.

It seems the AWS Credentials Custom Endpoints gets ignored even when configured, it also seems that a custom url gets created to interact instead of using a correct s3 url.

 used: 
 https://images.example.com.s3.eu-west-1.amazonaws.com/
 https://{bucket}.s3.{region}.amazonaws.com/
 
 expected: 
 https://s3.eu-west-1.amazonaws.com/images.example.com/
 https://s3.{region}.amazonaws.com/{bucket}/

net result of this - ERR_TLS_CERT_ALTNAME_INVALID 

```
NodeApiError: Hostname/IP does not match certificate's altnames: Host: images.example.com.s3.eu-west-1.amazonaws.com. is not in the cert's altnames: DNS:s3-eu-west-1.amazonaws.com, DNS:*.s3-eu-west-1.amazonaws.com, DNS:s3.eu-west-1.amazonaws.com, DNS:*.s3.eu-west-1.amazonaws.com, DNS:s3.dualstack.eu-west-1.amazonaws.com, DNS:*.s3.dualstack.eu-west-1.amazonaws.com, DNS:*.s3.amazonaws.com, DNS:*.s3-control.eu-west-1.amazonaws.com, DNS:s3-control.eu-west-1.amazonaws.com, DNS:*.s3-control.dualstack.eu-west-1.amazonaws.com, DNS:s3-control.dualstack.eu-west-1.amazonaws.com, DNS:*.s3-accesspoint.eu-west-1.amazonaws.com, DNS:*.s3-accesspoint.dualstack.eu-west-1.amazonaws.com, DNS:*.s3-deprecated.eu-west-1.amazonaws.com, DNS:s3-deprecated.eu-west-1.amazonaws.com, DNS:s3-external-3.amazonaws.com, DNS:*.s3-external-3.amazonaws.com
    at Object.requestWithAuthentication (/usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/NodeExecuteFunctions.js:1187:19)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Object.requestWithAuthentication (/usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/NodeExecuteFunctions.js:1876:20)
    at Object.awsApiRequest (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/S3/V1/GenericFunctions.js:25:12)
    at Object.awsApiRequestSOAP (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/S3/V1/GenericFunctions.js:39:22)
    at Object.execute (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/S3/V1/AwsS3V1.node.js:504:40)
    at Workflow.runNode (/usr/local/lib/node_modules/n8n/node_modules/n8n-workflow/dist/Workflow.js:700:19)
    at /usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/WorkflowExecute.js:656:53
    at /usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/WorkflowExecute.js:1058:20
    
```
